### PR TITLE
Add modes support for RestApi

### PR DESCRIPTION
**Pull request by c2tarun**
*Issue #, if available:*

*Description of changes:*
Adding support for [Mode](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-restapi.html#cfn-apigateway-restapi-mode) property added by API Gateway.

*Description of how you validated changes:*
Unit tests
Integration test are using `overwrite` mode value to ensure that when an path is removed it gets removed from API Gateway as well.

*Checklist:*

- [x] Write/update tests
- [x] `make pr` passes
- [ ] Update documentation
- [x] Verify transformed template deploys and application functions as expected

*Examples?*

Please reach out in the comments, if you want to add an example. Examples will be 
added to `sam init` through https://github.com/awslabs/aws-sam-cli-app-templates/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by awood45 — APPROVED** [review]

File: integration/helpers/base_test.py [hunk 1]
```diff
@@ -146,6 +147,28 @@ def create_and_verify_stack(self, file_name, parameters=None):
...
+
+        Parameters
+        ----------
+        file_name : string
+            Template file name
+        parameters : list
+            List of parameters
+        """
+        if not self.stack_name:
+            raise Exception("Stack not created.")
```
Comment (sriram-mv) [inline integration/helpers/base_test.py:163]: Typed Execptions?

**Review by sriram-mv — APPROVED** [review]
Overall Looks good.

File: pip-wheel-metadata/aws_sam_translator.dist-info/LICENSE [hunk 2]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (mndeveci) [inline pip-wheel-metadata/aws_sam_translator.dist-info/LICENSE:1]: Why we added license file and couple of more to the repo? Was adding `pip-wheel-metada` intended?

**Review by mndeveci — CHANGES_REQUESTED** [review]
Implementation LGTM, but I was wondering why we added `pip-wheel-metadata` folder into project?

File: pip-wheel-metadata/aws_sam_translator.dist-info/LICENSE (hunk 2 above)
Comment (c2tarun) [reply pip-wheel-metadata/aws_sam_translator.dist-info/LICENSE]: Fixed it. It was side-effect of `git add .`

**Review by mndeveci — APPROVED** [review]

**Pull request merged**